### PR TITLE
Fix typing in the application

### PR DESCRIPTION
**Pull request by fridex**
## This introduces a breaking change

- [x] No

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -73,12 +76,12 @@
 )
 
 
-def _buildlog_metadata(api_endpoint: str = None, build_log: str = None) -> dict:
+def _buildlog_metadata(api_endpoint: str = Optional[None], build_log: str = Optional[None]) -> Dict[str, str]:
     """Gather metadata for the build log."""
     # Update the metadata with more details
     if not build_log:
         return {}
-    return {"apiversion": api_endpoint, "kind": "BuildLog", "log": build_log, "metadata": "string"}
```
Comment (fridex) [inline app.py:81]: Removal of the metadata string is intentional here. It has no effect.

File: app.py [hunk 2]
```diff
@@ -73,12 +76,12 @@
 )
 
 
-def _buildlog_metadata(api_endpoint: str = None, build_log: str = None) -> dict:
+def _buildlog_metadata(api_endpoint: str = Optional[None], build_log: str = Optional[None]) -> Dict[str, str]:
```
Comment (KPostOffice) [inline app.py]: shouldn't these be `foo: Optional[str] = None`?

Comment (KPostOffice) [reply app.py]: Otherwise their default value will be set to a type annotation

File: app.py [hunk 3]
```diff
@@ -257,13 +262,13 @@ def _do_analyze_build(
 def _push_image(
     image: str,
     push_registry: str,
-    src_registry_user: str = None,
-    src_registry_password: str = None,
-    dst_registry_user: str = None,
-    dst_registry_password: str = None,
+    src_registry_user: str = Optional[None],
+    src_registry_password: str = Optional[None],
```
Comment (KPostOffice) [inline app.py]: See above comment

File: app.py [hunk 4]
```diff
@@ -489,20 +496,20 @@ def _submitter(
...
-    src_registry_password: str = None,
-    dst_registry_user: str = None,
-    dst_registry_password: str = None,
-    push_registry: str = None,
-    analyze_existing: bool = None,
-    workers_count: int = None,
-    environment_type: str = None,
+    src_registry_user: str = Optional[None],
+    src_registry_password: str = Optional[None],
+    dst_registry_user: str = Optional[None],
```
Comment (KPostOffice) [inline app.py]: Here also

**Review by KPostOffice — COMMENTED** [review]
Just a few changes I think. Type hints look a bit wonky.

File: app.py (hunk 2 above)
Comment (fridex) [reply app.py]: Yes, silly mistake (tired eyes? :)) Created typing fix for this typing fix in 6f04ae8, PTAL

**Comment by fridex** [comment]
/approve

@KPostOffice I will merge this one to continue with the work. Feel free to raise any comments if you have.

**Pull request merged**